src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Projects/CpsPackageReferenceProject.cs

```diff
@@ -221,65 +199,105 @@ private void IgnoreUnsupportProjectReference(PackageSpec project)
 
         #region NuGetProject
 
+        /// <summary>
+        /// Gets the installed (top level) package references for this project. 
+        /// </summary>
         public async override Task<IEnumerable<PackageReference>> GetInstalledPackagesAsync(CancellationToken token)
+        {
+            var packages = await GetAllPackagesAsync(token);
```

Comment:
This ends up getting both installed and transitive packages but then discarding the unwanted transitive packages.  Is there a perf regression here?  Should we only retrieve installed packages here?

Reply:
I think the idea is that calculating the transitive dependencies should not be super expensive, and that this would only take a perf hit whenever the assets file needs to be read. 

So we're betting getting the transitive dependencies to be called frequently enough where the slight overhead it might add to GetInstalledPackages is better than reading the assets file twice. 

Chatted with @birgithi and @rrelyea and I suggested that we measure it so we can make a more educated decision.

Reply:
I ran some tests and found that most of the time goes into reading the assets file compared to finding the package dependencies. So I think it makes sense to read the assets file only once to get both installed and transitive packages.